biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java

```diff
@@ -1210,4 +1210,24 @@ public void removePrefix(String prefixLow) {
 		directories.subMap(prefixLow, prefixHigh)
 			.clear();
 	}
+
+	public void removeSubDirs(String path) {
+		if (path.endsWith("/")) {
+			path = path.substring(0, path.length() - 1);
+		}
+
+		Map<String, Resource> dir = directories.get(path);
+
+		if ((dir == null) || dir.isEmpty()) {
+			return;
+		}
+
+		dir.keySet()
+			.forEach(s -> {
```

Comment:
Can't this be simplified to 

```java
dir.keySet().forEach(this::removePrefix);
```

I cannot imagine the null check is needed and emptiness would be handled anyway.

Reply:
the `isEmpty` is not needed, but the null check is.

Reply:
Why would the keySet have null values in it?